grade/result.py

```diff
@@ -61,6 +61,11 @@ def getExceptions(self, test) -> List[Tuple]:
         return [m for f, m in [*self.failures, *self.errors] if f == test]
 
     def getName(self, test):
+        # If a user-defined name exists, use it.
+        name = self.getattr(test, "__testname__", None)
+        if name:
```

Comment:
Let's avoid having multiple return statements. Change this to `if not name` and embed our other logic in that block. Return name at the very bottom after all of this.

Reply:
Fixed!